### PR TITLE
Run prettier on all markdown files

### PR DESCRIPTION
**Pull request by feuGeneA**
-   [x] Update package-level scripts to have `yarn lint` check javascript files in all cases where `yarn lint:fix` would fix them, per [this review comment](https://github.com/nomiclabs/hardhat/pull/1582#issuecomment-866924284).  Addressed by https://github.com/nomiclabs/hardhat/pull/1582/commits/b7ecef1b7e733673371e329a74d9b97e6277e0f5

### CODE REVIEW

**Comment by fvictorio** [comment]
So, if I'm understanding this correctly, then...

- Running `yarn lint` from the root will run the `lint` script in each package **and** check the markdown files. Similarly, running `yarn lint:fix` will run the `lint:fix` script in each package and format all the markdown files.
- Running `yarn lint` in an individual package wasn't modified: it just runs `tslint` in the source and test directories.
- Running `yarn lint:fix` in an individual package will run the `lint` script with the `--fix` flag. In some packages, it will also format the javascript files inside the test directory.

It seems to me that this is missing a piece: if `lint:fix` formats some javascript files, then `lint` should also check them.

**Comment by alcuadrado** [comment]
I implemented the following changes:

* Moved the prettier scripts to each package, so that they are independent.
* Added lint and lint:fix to `docs/`
* I also added `.prettierignore` files so that we can run `prettier **/*.{extensions}` and make sure that we are linting/autoformating everything.
* Added `json` to the list of files that we lint and autoformat.
* Enabled `--prose-wrap=never` in prettier, which formats markdown paragraphs as a single line (i.e. no hard line wrapping)
* Created a lint docs github actions job

**Comment by alcuadrado** [comment]
> * Running `yarn lint` from the root will run the `lint` script in each package **and** check the markdown files. Similarly, running `yarn lint:fix` will run the `lint:fix` script in each package and format all the markdown files.

Now, `yarn lint` in the root will run `lint` in each package. Same for `lint:fix`.

> * Running `yarn lint` in an individual package wasn't modified: it just runs `tslint` in the source and test directories.

It now also runs `prettier --check`.

> * Running `yarn lint:fix` in an individual package will run the `lint` script with the `--fix` flag. In some packages, it will also format the javascript files inside the test directory.

It now runs `prettier --write` and `tslint --fix` in every package, except `common`, which doesn't use `tslint`.

> It seems to me that this is missing a piece: if `lint:fix` formats some javascript files, then `lint` should also check them.

This is how it works now.

File: packages/e2e/package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   "author": "Nomic Labs LLC",
   "license": "MIT",
   "scripts": {
-    "lint:fix": "prettier --write \"test/**/*.js\" && yarn lint --fix",
-    "lint": "prettier --check \"test/**/*.js\" && tslint --config tslint.json --project ./tsconfig.json",
+    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
+    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/e2e/package.json]: I think you accidentally dropped the `--fix` during the switch here.

File: packages/hardhat-docker/package.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
     "docker"
   ],
   "scripts": {
-    "lint:fix": "yarn lint --fix",
-    "lint": "tslint --config tslint.json --project ./tsconfig.json",
+    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
+    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-docker/package.json]: I think you dropped the `--fix` in the switch here.

File: docs/package.json [hunk 3]
```diff
@@ -2,8 +2,8 @@
   "scripts": {
     "dev": "vuepress dev .",
     "build": "NODE_ENV=production vuepress build .",
-    "lint": "prettier --check \"**/*.{js,md,json}\"",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\""
+    "lint": "prettier --prose-wrap=never --check \"**/*.{js,md,json}\"",
```
Comment (feuGeneA) [inline docs/package.json:5]: I know we like to say that we use prettier without any configuration, but now that we've got two pieces of config information (prose-wrap and filetypes) that are identical across all of the packages, I'm wondering if it wouldn't be nice to simplify things and just use a config file.

File: packages/hardhat-docker/package.json [hunk 4]
```diff
@@ -13,8 +13,10 @@
     "docker"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-docker/package.json:19]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-etherscan/package.json [hunk 5]
```diff
@@ -20,8 +20,10 @@
     "etherscan"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-etherscan/package.json:26]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-ganache/package.json [hunk 6]
```diff
@@ -17,8 +17,10 @@
     "testing-network"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-ganache/package.json:23]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-shorthand/package.json [hunk 7]
```diff
@@ -17,8 +17,10 @@
     "cli"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-shorthand/package.json:23]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-solhint/package.json [hunk 8]
```diff
@@ -18,8 +18,10 @@
     "linter"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-solhint/package.json:24]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-solpp/package.json [hunk 9]
```diff
@@ -18,8 +18,10 @@
     "solpp"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-solpp/package.json:24]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-truffle4/package.json [hunk 10]
```diff
@@ -17,8 +17,10 @@
     "truffle-contract"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-truffle4/package.json:23]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-truffle5/package.json [hunk 11]
```diff
@@ -17,8 +17,10 @@
     "truffle-contract"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-truffle5/package.json:23]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-vyper/package.json [hunk 12]
```diff
@@ -16,8 +16,10 @@
     "vyper"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-vyper/package.json:22]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-waffle/package.json [hunk 13]
```diff
@@ -16,8 +16,10 @@
     "ethers.js"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-waffle/package.json:22]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-web3-legacy/package.json [hunk 14]
```diff
@@ -17,8 +17,10 @@
     "web3.js"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-web3-legacy/package.json:23]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-web3/package.json [hunk 15]
```diff
@@ -17,8 +17,10 @@
     "web3.js"
   ],
   "scripts": {
-    "lint": "prettier --check \"**/*.{js,md,json}\" && tslint --config tslint.json --project ./tsconfig.json",
-    "lint:fix": "prettier --write \"**/*.{js,md,json}\" && tslint --fix --config tslint.json --project ./tsconfig.json",
+    "lint": "yarn prettier --check && yarn tslint",
+    "lint:fix": "yarn prettier --write && yarn tslint --fix",
+    "prettier": "prettier \"**/*.{js,md,json}\"",
+    "tslint": "tslint --fix --config tslint.json --project ./tsconfig.json",
```
Comment (feuGeneA) [inline packages/hardhat-web3/package.json:23]:
```suggestion
    "tslint": "tslint --config tslint.json --project ./tsconfig.json",
```

File: packages/hardhat-docker/package.json (hunk 4 above)
Comment (alcuadrado) [reply packages/hardhat-docker/package.json:19]: Ouch, sorry I made so many silly mistakes in this PR.

**Review by fvictorio — APPROVED** [review]
LGTM but please check the commit I added

**Comment by feuGeneA** [comment]
> LGTM but please check the commit I added

LGTM

**Pull request merged**